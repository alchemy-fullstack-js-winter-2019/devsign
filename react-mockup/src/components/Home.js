import React from 'react';
import Header from './app/Header';
import Trending from '../containers/Trending';
import Footer from './app/Footer';

function Home() {
  return (
    <>
    <Header />
    <main>
      <Trending />
    </main>
    <Footer />
    </>
  );
}

export default Home;
