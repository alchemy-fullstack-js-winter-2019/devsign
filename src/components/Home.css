.Home {
  display: grid;
  grid-gap: 10px;
  grid-template-columns: 150px 150px 150px;
  padding: 1 em;
  margin: 20px;
}
header {
  grid-column: 1 / span 4;
  background-color: #84B3DA;
}
footer {
  position: absolute;
  bottom: 0;
}
.sidebar {
  background-color: #84B3DA;
}