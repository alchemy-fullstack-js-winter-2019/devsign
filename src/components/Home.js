import React from 'react';

export default function Home() {
  return (
    <>
      <section>
        <figure>
          <img src="https://pbs.twimg.com/media/C8QsNInXUAAyjZQ.jpg" alt="user avatar" />
          <h3>T_on_A</h3>
        </figure>
      </section>
      <section>
        <label>Post a Tweet</label>
        <input type="text" />
        <button>Tweet</button>
        <button>Add An Image</button>
        <button>Add A Gif</button>
        <button>Add Location</button>
      </section>
      <section>
        <ol>
          <li>List of tweets from users you are following</li>
        </ol>
      </section>
      <aside>
        <ul>
          <section>
            <figure>
              <img src="https://pbs.twimg.com/media/C8QsNInXUAAyjZQ.jpg" alt="user avatar" />
            </figure>
            <h3>Jill Hill</h3>
            <p>Todays lab sucks</p>
          </section>
          <section>
            <figure>
              <img src="https://pbs.twimg.com/media/C8QsNInXUAAyjZQ.jpg" alt="user avatar" />
            </figure>
            <h3>Jack Hill</h3>
            <p>Todays lab is AMAZING</p>
          </section>
          <section>
            <figure>
              <img src="https://pbs.twimg.com/media/C8QsNInXUAAyjZQ.jpg" alt="user avatar" />
            </figure>
            <h3>Hilly Jill</h3>
            <p>Todays lab is all uphill</p>
          </section>
        </ul>
      </aside>
    </>
  );
}
