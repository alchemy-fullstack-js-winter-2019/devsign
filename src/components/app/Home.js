import React, { Fragment } from 'react';
import Header from './Header';
import Trending from '../chirps/Trending';
import ChirpsContainer from '../../containers/PopularChirpsContainer';
import MiniProfileContainer from '../../containers/MiniProfileContainer';
import { Div1, Div2, Section } from './HomeStyles';

export default function DesktopHome() {
  return (
    <Fragment>
      <Header />
      <Section>
        <Div1>
          <MiniProfileContainer />
          <Trending />
        </Div1>
        <Div2>
          <ChirpsContainer />
        </Div2>
      </Section>
    </Fragment>
  );
}
