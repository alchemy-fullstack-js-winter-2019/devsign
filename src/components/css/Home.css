.feed {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 20px;

  h2 {
    color: red;
  }
}
