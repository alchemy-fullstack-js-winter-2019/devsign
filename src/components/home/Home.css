
img {
  max-height: 20vh;
  border-radius: 60px
}
ul {
  padding: 0;
}
.profile  {
  display: none;
  img {
    height: 10vh;
  }
  h3 {
    font-size: 1em;
  }
}

.post {
  margin: 2px;
  background-color: #DEEBED;
  button {
    background-color: #D8D8D8;
  }
  input {
    border: 1px solid #D8D8D8;
  }
}

