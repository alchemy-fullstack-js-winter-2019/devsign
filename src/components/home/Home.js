import React from 'react';
import styles from './Home.css';
import Tweets from '../tweets/Tweets';

const popularTweets = [
  { _id: '0001', text: 'My first tweet', user: { handle: 'Tweedy', userPic: 'http://i.pravatar.cc/1000' } },
  { _id: '0002', text: 'I just flew in from Boise', user: { handle: 'Mocking Jay', userPic: 'http://i.pravatar.cc/1000' } },
  { _id: '0003', text: 'A guy is sitting at an airport bar', user: { handle: 'Parrot', userPic: 'http://i.pravatar.cc/1000' } },
  { _id: '0004', text: '...Tower, please call me a fuel truck.', user: { handle: 'Doug', userPic: 'http://i.pravatar.cc/1000' } },
  { _id: '0005', text: 'KLM 242 expect start up in two hours', user: { handle: 'Polly', userPic: 'http://i.pravatar.cc/1000' } }
];

export default function Home() {
  return (
    <main 
      className={styles.Home}
    > 
      <Tweets tweets={popularTweets} />
    </main>
  );
}
