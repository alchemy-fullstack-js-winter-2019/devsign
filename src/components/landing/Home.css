.main {
  ul {
    padding: 0;
  }
  .profile  {
    display: none;
    img {
      height: 10vh;
    }
    h3 {
      font-size: 1em;
    }
    img {
      max-height: 20vh;
      border-radius: 60px
    }
  }
  
  .post {
    margin: 2px;
    background-color: #DEEBED;
    button {
      background-color: #D8D8D8;
    }
    input {
      border: 1px solid #D8D8D8;
    }
  }
  
  @media screen and (min-width 1000px) {
    .profile {
      
    }
  }
}  
