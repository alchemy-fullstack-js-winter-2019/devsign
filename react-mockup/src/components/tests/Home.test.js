import React from 'react';
import Home from '../Home';
import { MemoryRouter } from 'react-router';
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Home snapshot', () => {
  it('can match a snapshot', () => {
    const tree = shallow(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );
    expect(tree).toMatchSnapshot();
  });
});
