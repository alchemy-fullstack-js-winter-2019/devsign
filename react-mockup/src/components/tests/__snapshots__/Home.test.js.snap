// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot can match a snapshot 1`] = `
Array [
  <header>
    <img
      alt="Hummingbird logo"
      src="react-mockup/src/assets/logo.png"
    />
    <nav>
      <p>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </p>
      <p>
        <a
          href="/hum-feed"
          onClick={[Function]}
        >
          HumFeed
        </a>
      </p>
      <p>
        <a
          href="/user"
          onClick={[Function]}
        >
          Example User
        </a>
      </p>
    </nav>
  </header>,
  <main>
    <section>
      <h2>
        Log-in
      </h2>
      <label
        htmlFor="username"
      >
        Username:
      </label>
      <input
        name="username"
        type="text"
      />
      <label
        htmlFor="password"
      >
        Password:
      </label>
      <input
        name="password"
        type="password"
      />
      <button>
        Sign-in
      </button>
    </section>
    <section>
      <h3>
        What is trending...
      </h3>
      <ol>
        <li>
          Top Hums go here
        </li>
        <li>
          Top Hums go here
        </li>
        <li>
          Top Hums go here
        </li>
        <li>
          Top Hums go here
        </li>
        <li>
          Top Hums go here
        </li>
      </ol>
    </section>
  </main>,
  <footer>
    <p>
      Â© Hummingbird, 2019
    </p>
  </footer>,
]
`;
